### PR TITLE
Feature/channel growth per day

### PR DESCRIPTION
**Pull request by eshleebien**

### CODE REVIEW

File: commands/channel_growth.js [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+        network = {},
+
+        start = function () {
+            app.log.info('Fetching ...');
+
+            query = "SELECT distinct(channel_id)" +
+                " FROM jobs ";
+
+            if (app.argv.limit && app.argv.limit > 0) {
+                limit = app.argv.limit;
```
Comment (wildtron) [inline commands/channel_growth.js]: `+app.argv.limit` so that you get `int` la..

File: commands/channel_growth.js [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+                app.log.error(err);
+                p_count = 1;
+
+                return done();
+            }
+
+            if (!rows.length) {
+                app.log.info('No channels to update');
+
+                process.exit();
```
Comment (adin234) [inline commands/channel_growth.js]: why do we call `process.exit()` T_T

File: commands/channel_growth.js [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+
+            if (err) {
+
+                if (err.status_code >= 500) {
+                    return request.retry();
+                }
+
+                if (err.status_code == undefined) {
+                    needs_retry.push(channel.channel_id)
+                    console.log(err);
```
Comment (adin234) [inline commands/channel_growth.js]: be consistent with `app.log` and 'console.log`

File: commands/channel_growth.js [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+                });
+
+                failed++;
+
+                channel.bar.tick();
+
+                return done();
+            }
+
+            if (result && result.rows && result.rows[0]) {
```
Comment (adin234) [inline commands/channel_growth.js]: why do we check only for `result.rows[0]` did you mean `result.rows.length`?

File: commands/channel_growth.js [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+            }
+            else {
+                failed++;
+
+                return done();
+            }
+        },
+
+        done = function () {
+            p_count --;
```
Comment (adin234) [inline commands/channel_growth.js]: you can remove the empty space ` ` here before `--`

File: commands/channel_growth.js [hunk 6]
```diff
@@ -0,0 +1,285 @@
...
+
+            while(length --) {
+                ids.push(channels[length].channel_id);
+            }
+
+            needs_retry.forEach(function (c_id) {
+                ids.splice(ids.indexOf(c_id), 1);
+            });
+
+            console.log(ids.length);
```
Comment (adin234) [inline commands/channel_growth.js]: be consistent with `app.log` and `console.log`

File: commands/channel_growth.js [hunk 7]
```diff
@@ -0,0 +1,285 @@
...
+    moment = require('moment'),
+    ProgressBar = require('progress'),
+    app = {};
+
+module.exports = function (cb) {
+
+    var app = this,
+        limit = 20,
+        offset = 0,
+        query = '',
```
Comment (pprmint) [inline commands/channel_growth.js]: no need to declare `query` globally

File: commands/channel_growth.js [hunk 8]
```diff
@@ -0,0 +1,287 @@
...
+        }),
+
+        start = function () {
+            var query = '';
+
+            app.log.info('Fetching ...');
+
+            sqs.receiveMessage({
+                //QueueUrl : app.config.get('AWS').SQS.QUEUES.jobs,
+                QueueUrl : 'https://sqs.us-east-1.amazonaws.com/813913971945/test_queue',
```
Comment (adin234) [inline commands/channel_growth.js]: you really removed this from the config?

Comment (eshleebien) [inline commands/channel_growth.js]: ah sorry i forgot to remove our test queue

File: commands/channel_growth.js [hunk 9]
```diff
@@ -0,0 +1,287 @@
...
+                if (!channel.records.length) {
+                    return done();
+                }
+
+                while (days_length --) {
+                    days.push(channel.records[days_length][0]);
+                }
+
+                channel.date_list.forEach(function (d) {
+                    if (days.indexOf(d) === -1) {
```
Comment (adin234) [inline commands/channel_growth.js]: we can use `~` here

File: commands/channel_growth.js [hunk 10]
```diff
@@ -0,0 +1,287 @@
...
+                        channel.records.push([moment(d).format('YYYY-MM-DD'), 0, 0, 0, 0]);
+                        //days.push(moment(d).format('YYYY-MM-DD')); //debugging purpose
+                    }
+                });
+
+                days.forEach(function (d) {
+                    if (channel.date_list.indexOf(d) === -1) {
```
Comment (adin234) [inline commands/channel_growth.js]: we can use `!~` here

**Comment by eshleebien** [comment]
@adin234 please check again :)

**Comment by adin234** [comment]
hello, we have updated javascript standards just this weekend, we think it is best if we move the coding style of this repo to the updated [standards](https://github.com/anyTV/JS-conventions)

**Comment by eshleebien** [comment]
@adin234 done

File: commands/channel_growth.js [hunk 11]
```diff
@@ -0,0 +1,285 @@
...
+        needs_retry = [],
+        network = {},
+        sqs = new aws.SQS({
+            accessKeyId : app.config.get('AWS').SQS.KEY,
+            secretAccessKey : app.config.get('AWS').SQS.SECRET,
+            region : app.config.get('AWS').SQS.REGION
+        });
+
+        function start () {
+            var query = '';
```
Comment (wildtron) [inline commands/channel_growth.js]: where was this used?

File: commands/channel_growth.js [hunk 12]
```diff
@@ -0,0 +1,285 @@
...
+                MaxNumberOfMessages : +app.config.get('AWS').SQS.RETRIEVED_MESSAGE,
+                MessageAttributeNames : ['All'],
+                VisibilityTimeout : 30,
+            }, authenticate);
+        };
+
+        function authenticate (err, data) {
+            if (err) {
+                app.log.error(err);
+                p_count = 1;
```
Comment (wildtron) [inline commands/channel_growth.js]: where is this used?

File: commands/channel_growth.js [hunk 13]
```diff
@@ -0,0 +1,285 @@
...
+                    client_id : app.config.get('google_api')[0].client_id,
+                    client_secret : app.config.get('google_api')[0].client_secret,
+                    refresh_token : app.config.get('google_api')[0].refresh_token,
+                    grant_type : 'refresh_token'
+                })
+                .then(get_dates);
+        };
+
+        function get_dates (err, result) {
+            var query = '';
```
Comment (wildtron) [inline commands/channel_growth.js]: where is this used inside this function?

File: commands/channel_growth.js [hunk 14]
```diff
@@ -0,0 +1,285 @@
...
+                    total: channel.record_length
+                });
+
+                days_length = 0;
+                channel.records.forEach(function (row) {
+                    if (days_length) {
+                        channel.records[days_length][1] += channel.records[days_length - 1][1]; // aggregate views
+                        channel.records[days_length][2] += channel.records[days_length - 1][2]; // aggregate subs
+                        channel.records[days_length][4] += channel.records[days_length - 1][4]; // aggregate comments
+                        channel.records[days_length][2] -= channel.records[days_length][3]; //subs_gained - subs_lost
```
Comment (wildtron) [inline commands/channel_growth.js]: combine statements that use similar index

**Pull request merged**